### PR TITLE
Powerstore support for dell csm operator

### PR DESCRIPTION
**Pull request by karthikk92**
# Description
Powerstore support for dell csm operator

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/613 |

Note :  
- External health monitor is not working and will be taken care in another PR
- Unit tests are failing now and unit test and e2e tests will be added in upcoming PR.
- Updated allowed list CVE-2019-1010022

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Installed latest powerstore driver and working fine.
test

### CODE REVIEW

File: samples/storage_csm_powerstore_v250.yaml [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  name: test-powerstore
+  namespace: test-powerstore
+spec:
+  driver:
+    csiDriverType: "powerstore"
+    csiDriverSpec:
+      # fsGroupPolicy: Defines if the underlying volume supports changing ownership and permission of the volume before being mounted.
+      # Allowed values: ReadWriteOnceWithFSType, File , None
+      # Default value: ReadWriteOnceWithFSType
+      fSGroupPolicy: "File"
```
Comment (rensyct) [inline samples/storage_csm_powerstore_v250.yaml:13]: @kk3, please ensure that changes made in samples are reflected properly as i see there is hardcoding currently

**Review by rensyct — COMMENTED** [review]
@kk3, please create a JIRA ticket to ensure that any changes made to the samples are reflected and field is set as per what is set in the samples

**Comment by karthikk92** [comment]
> @kk3, please create a JIRA ticket to ensure that any changes made to the samples are reflected and field is set as per what is set in the samples

Created story. https://jira.cec.lab.emc.com/browse/KRV-11170

**Review by panigs7 — DISMISSED** [review]

File: config/samples/storage_v1_csm_powerstore.yaml [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+          value: "true"
+
+      # nodeSelector: Define node selection constraints for controller pods.
+      # For the pod to be eligible to run on a node, the node must have each
+      # of the indicated key-value pairs as labels.
+      # Leave as blank to consider all nodes
+      # Allowed values: map of key-value pairs
+      # Default value: None
+      nodeSelector:
+      # Uncomment if nodes you wish to use have the node-role.kubernetes.io/master taint
```
Comment (rensyct) [inline config/samples/storage_v1_csm_powerstore.yaml:75]: Please remove examples of master from node selection and tolerations

File: config/samples/storage_v1_csm_powerstore.yaml [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        # CHAP password will be autogenerated by driver
+        - name: "X_CSI_POWERSTORE_ENABLE_CHAP"
+          value: "true"
+        # X_CSI_HEALTH_MONITOR_ENABLED: Enable/Disable health monitor of CSI volumes from node plugin - volume usage
+        # Allowed values:
+        #   true: enable checking of health condition of CSI volumes
+        #   false: disable checking of health condition of CSI volumes
+        # Default value: false
+        - name: X_CSI_HEALTH_MONITOR_ENABLED
+          value: "true"
```
Comment (rensyct) [inline config/samples/storage_v1_csm_powerstore.yaml:104]: Set this to the default value

File: samples/storage_csm_powerstore_v250.yaml [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+          value: "false"
+
+      # nodeSelector: Define node selection constraints for node pods.
+      # For the pod to be eligible to run on a node, the node must have each
+      # of the indicated key-value pairs as labels.
+      # Leave as blank to consider all nodes
+      # Allowed values: map of key-value pairs
+      # Default value: None
+      nodeSelector:
+      # Uncomment if nodes you wish to use have the node-role.kubernetes.io/master taint
```
Comment (rensyct) [inline samples/storage_csm_powerstore_v250.yaml:113]: Remove examples of master from node selector and tolerations

File: config/samples/storage_v1_csm_powerstore.yaml (hunk 3 above)
Comment (karthikk92) [reply config/samples/storage_v1_csm_powerstore.yaml]: Updated and taken care

File: samples/storage_csm_powerstore_v250.yaml (hunk 4 above)
Comment (karthikk92) [reply samples/storage_csm_powerstore_v250.yaml]: Updated and taken care

**Comment by rensyct** [comment]
@karthikk92 , please check the failure for formatting, linting and vetting workflow

**Comment by karthikk92** [comment]
> @karthikk92 , please check the failure for formatting, linting and vetting workflow



> @karthikk92 , please check the failure for formatting, linting and vetting workflow



> @karthikk92 , please check the failure for formatting, linting and vetting workflow

taken care

**Review by panigs7 — APPROVED** [review]

**Review by rajendraindukuri — APPROVED** [review]

**Comment by rensyct** [comment]
@karthikk92 , please mention here JIRA ticket ID that is created for unit tests as the check is failing in this PR

**Comment by karthikk92** [comment]
> * Unit tests are failing now and unit test and e2e tests will be added in upcoming PR.



> @karthikk92 , please mention here JIRA ticket ID that is created for unit tests as the check is failing in this PR



> @karthikk92 , please mention here JIRA ticket ID that is created for unit tests as the check is failing in this PR

Jira id : https://jira.cec.lab.emc.com/browse/KRV-7639

**Review by rensyct — APPROVED** [review]

**Pull request merged**